src/write_sensu.c

```diff
@@ -39,6 +39,53 @@
 #include <stddef.h>
 
 #include <stdlib.h>
+#ifdef HAVE_KERNEL_SOLARIS
```

Comment:
The problem is that I need a critical ZFS update for the SSD Cache Configuration I run on the buildfarm which is only available on this Kernel Update Release. However, most users run older Solaris Kernel Releases, so I restrict the libc linker contract. `configure` correctly detects `asprintf` to be non-available, but the header files obviously include it.

If you include
```
#ifdef KERNEL_SOLARIS
#ifndef HAVE_ASPRINTF
#define asprintf _asprintf
#define vasprintf _vasprintf
#endif
#endif
```
before all other `#include`
and then
```
#undef vasprintf
#undef asprintf
```
in the Solaris asprintf substitute section it works.

But I could also understand that this is a OpenCSW special case and I would be willing to maintain this patch downstream as it is probably not helpful for everyone.